org.osate.ui/src/org/osate/ui/wizards/NewAadlPackageWizard.xtend

```diff
@@ -67,6 +80,25 @@ class NewAadlPackageWizard extends AbstractNewFileWizard {
 			graphicalButton.setSelection(false)
 		]
 	}
+		
+	// Not going to get here if no project is selected
+	override String validateFileName(IContainer parent, String packageName) {
+		/* Parent might be a Project, which causes problems below, so let's append
+		 * a bogus folder to it.
+		 */
+		val IFolder fakeFolder = parent.getFolder(Path.forPosix(".fake"))
+		val Resource rsrc = OsateResourceUtil.getResource(fakeFolder)
+		val scope = globalScopeProvider.getScope(
+			rsrc, Aadl2Package.eINSTANCE.getPackageRename_RenamedPackage(), null
+		)
+		val qualifiedName = qNameConverter.toQualifiedName(packageName);
+		if (scope.getSingleElement(qualifiedName) !== null) {
+			return "Package '" + packageName + "' already exists in scope."
```

Comment:
The error message is correct but not very useful. It should read: "Package xyz already exists in project abc (file fgh.aadl)." This helps the user find the other package.